hs_tracking/models.py

```diff
@@ -18,31 +18,38 @@
 
 
 class SessionManager(models.Manager):
-    def for_request(self, request):
+    def for_request(self, request, user=None):
+        if hasattr(request, 'user'):
+            user = request.user
+
         signed_id = request.session.get('hs_tracking_id')
         if signed_id:
             tracking_id = signing.loads(signed_id)
             cut_off = datetime.now() - timedelta(seconds=SESSION_TIMEOUT)
             session = None
+
             try:
                 session = Session.objects.filter(
                     variable__timestamp__gte=cut_off).filter(id=tracking_id['id']).first()
             except Session.DoesNotExist:
                 pass
+
             if session is not None:
-                if session.visitor.user is None and request.user.is_authenticated():
+                if session.visitor.user is None and user.is_authenticated():
```

Comment:
@jbradberry Since you check on line 22 to assign user only when request object has user attribute, I assume it is possible that request does not have a user attribute, like an anonymous user perhaps? If that is the case, ```user``` object will be None, and this call ```user.is_authenticated()``` would fail.